src/Interpreters/Cache/LRUFileCachePriority.cpp

```diff
@@ -508,10 +508,9 @@ void LRUFileCachePriority::LRUIterator::incrementSize(size_t size, const CachePr
         cache_priority->log,
         "Incrementing size with {} in LRU queue for entry {}",
         size, entry->toString());
-
-    cache_priority->updateSize(size);
-    entry->size += size;
-
+    size_t prev_size = entry->getSize();
+    entry->increaseSize(size);
+    cache_priority->updateSize(entry->getSize() - prev_size);
```

Comment:
```suggestion
    /// We recalculate `diff_size` (not using `size` for `cache_priority` update)
    /// because if `use_read_cache_size = 1`, the actual size update could be different from `size` as a result of file size alignment.
    cache_priority->updateSize(entry->getSize() - prev_size);
```